### PR TITLE
refactor(react): Make ES6 imports of angular DI a little less ceremonious

### PR DESCRIPTION
**Pull request by christopherthielen**
Simplified exposing an angular DI token as an export.

Instead of: explicitly exporting a callback, importing and invoking it in `react.module`
Do this: import the `ReactInjector` and ask it for the injector.

```
export let ExecutionDetails: React.ComponentClass<IExecutionDetailsProps> = undefined;
ReactInjector.give(($injector: any) => ExecutionDetails = angular2react('executionDetails', new ExecutionDetailsComponent(), $injector) as any);
```

@jrsquared @anotherchrisberry PTAL

### CODE REVIEW

File: app/scripts/modules/core/react.module.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+export const REACT_MODULE = 'spinnaker.react';
+module(REACT_MODULE, [
+  'bcherny/ngimport',
+  STATE_EVENTS,
+]).run(function ($injector: any) {
+  // Make angular services importable and Convert angular components to react
+  ReactInjector.flush($injector);
+});
+
```
Comment (christopherthielen) [inline app/scripts/modules/core/react.module.ts]: no longer maintain an explicit master registry of "things that want the injector"

File: app/scripts/react.module.ts [hunk 2]
```diff
@@ -1,42 +0,0 @@
...
-}
-
-export const REACT_MODULE = 'spinnaker.react';
-module(REACT_MODULE, [
-  'bcherny/ngimport',
-  STATE_EVENTS,
-  UIROUTER_IMPORTS
-]).run(function ($injector: any) {
-  // Make angular services importable
-  StateEventsInject($injector);
```
Comment (christopherthielen) [inline app/scripts/react.module.ts]: no longer maintain this master registry of "things that want the injector"

**Comment by anotherchrisberry** [comment]
like

**Comment by jrsquared** [comment]
😍

**Comment by jrsquared** [comment]
LGREATTM

**Comment by christopherthielen** [comment]
> LGREATTM

Just realized what this acronym really means. 

I've always interpreted it as "let's get this merged" 🤦‍♂️

**Pull request merged**